server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -522,4 +544,26 @@ protected void validateOrderBy(String orderBy) throws IllegalArgumentException {
     public void updateLastLogonTime(String id, String zoneId) {
         jdbcTemplate.update(UPDATE_LAST_LOGON_TIME_SQL, timeService.getCurrentTimeMillis(), id, zoneId);
     }
+
+    private int getAllowedUserLimit(String zoneId) {
+        try {
+            UserConfig userConfig;
+            IdentityZone currentZone = identityZoneManager.getCurrentIdentityZone();
+            userConfig = (currentZone.getId().equals(zoneId)) ?
+                currentZone.getConfig().getUserConfig() :
+                jdbcIdentityZoneProvisioning.retrieve(zoneId).getConfig().getUserConfig();
+            return userConfig.getMaxUsers();
+        } catch (ZoneDoesNotExistsException e) {
+            logger.debug("could not retrieve identity zone with id: {}", zoneId);
+        }
+        return -1;
+    }
+
+    private void validateUserLimit(String zoneId) {
+        int maxAllowedUsers = getAllowedUserLimit(zoneId);
+        if (maxAllowedUsers > 0 && getZoneCount(zoneId, maxAllowedUsers) > (maxAllowedUsers -1)) {
```

Comment:
Please add a comment here that we compare against `maxAllowedUsers - 1` rather than `maxAllowedUsers` since we need capacity for the one user that is currently being created.